### DISCUSSION TITLE
Lesson 4: Explore Open Zeppelin, Initial supplu

### DISCUSSION THREAD

**Discussion by r0bchain** [Cyfrin/foundry-full-course-cu#1999] [General]
Hi there! 
I'm curious about why, when the initial supply is, for example, 1000 ether, the total of tokens in the owner account becomes 1000 NewTokenName when I import my new token in metamask. Does this mean that 1 ether is equal to 1 NewTokenName? What if I want to set the initial token value to, for example, 0.0005$ instead of 1 ether?
Thanks

**Comment by EngrPips** [comment]
Hello @r0bertinski, I understand your confusion; indeed, it is confusing if one doesn't properly understand it. So `ether` in a smart contract means `10^18`, so when you say your token initial supply is `1000 ether`, you get `1000` tokens. This means that the total supply of your token is `1000 x 10^18`. That is in terms of quantity, but now when you talk about value, i.e., what the token is actually worth, the price of 1 ether is only specific to the Ethereum token `Ether` and what 1 ether would be worth for your token mostly always depend on the pool where your token trade. If this does not make sense, please ask questions

**Reply by r0bchain** [reply]
Hmm, okay, so if my token name is "whateverToken" and I want to allocate an initial supply of 1000 tokens for the "whateverToken," I will always need to set the "initial supply" to 1000 ether, right?

Thanks!

**Reply by EngrPips** [reply]
Yeah, this is correct, considering your token decimal is `18`
